[Andong] iP

andong0909 opened the pull request:
# Woof! ChatBot

> "Your mind is for having ideas, not holding them." – David Allen

Woof! frees your mind of having to remember things you need to do. It's:

- Text-based
- Easy to learn
- ~FAST~ **SUPER FAST** to use

All you need to do is:

1. Download it from [here](#).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😊

And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Finding tasks with keywords
- [ ] Reminders (coming soon)

--- 

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Add greeting and exit message

Commit:
Add echo

Commit:
Add list

Commit:
Add Mark as Done

Commit:
Add ToDos, Events, Deadlines

Commit:
Add automated text UI testing

Commit:
Add error handling

Commit:
Add delete

Commit:
Delete unused variables

Commit:
Edit README file

Commit:
Add dummy enum

Commit:
Refractor to enum

Commit:
Add methods for file writing

Commit:
Add local storage of task lists

Commit:
Merge branch 'branch-level-7'

Commit:
Add date and time validation

Commit:
Add clear command

Commit:
Merge branch 'branch-level-8'

Commit:
Refractor by adding Command classes and others

Commit:
Refactor fully with parser and ui

Commit:
Refactor into packages

Commit:
Merge commit with gradle support

Commit:
Minor big fixes on importation of classes

Commit:
Delete commented out code snippet

Commit:
Fix minor bugs

Commit:
Configure gradle

Commit:
Add junit tests

Commit:
Configure to customise jar file name

Commit:
Update readme file

Commit:
Add javadoc comments

Commit:
Add find feature

nicholasdaijh commented on src/main/java/woof/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,27 @@
+public class AddDeadlineCommand extends Command {
+    /**
+     * Creates an instance of a command for adding deadlines.
+     * @param inputs
```
Consider describing what inputs represent to improve clarity for future maintainance of code

nicholasdaijh commented on src/main/java/woof/command/AddTodoCommand.java:
```diff
@@ -0,0 +1,27 @@
+public class AddTodoCommand extends Command {
+    /**
+     * Creates an instance of a command for adding todos.
+     * @param inputs
```
Consider describing what inputs represent to improve clarity for future maintainance of code

nicholasdaijh commented on src/main/java/woof/command/AddEventCommand.java:
```diff
@@ -0,0 +1,27 @@
+public class AddEventCommand extends Command {
+    /**
+     * Creates an instance of a command for adding events.
+     * @param inputs
```
Consider describing what inputs represent to improve clarity for future maintainance of code

nicholasdaijh commented on src/main/java/woof/Woof.java:
```diff
@@ -0,0 +1,70 @@
+        storage = new Storage();
+        try {
+            taskList = new TaskList(storage.loadTasks(filePath));
+        } catch (Exception e) {
```
Generic Exception is caught instead of handling specific exceptions. Consider catching more specific exceptions like IOException for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/Woof.java:
```diff
@@ -0,0 +1,70 @@
+                c.execute(taskList, ui);
+                Storage.saveTasks("data/tasks.txt");
+                isExit = c.isExit();
+            } catch (Exception e) {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,27 @@
+     * @throws Exception Possible exceptions.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws Exception {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/AddEventCommand.java:
```diff
@@ -0,0 +1,27 @@
+     * @throws Exception Possible exceptions.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws Exception {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/Command.java:
```diff
@@ -0,0 +1,22 @@
+        this.inputs = new String[0];
+    }
+
+    public abstract void execute(TaskList tasks, Ui ui) throws Exception;
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/DeleteCommand.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws Exception {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/FindCommand.java:
```diff
@@ -0,0 +1,16 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws Exception {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/MarkCommand.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws Exception {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws Exception {
```
Try to catch more specific exceptions for easier debugging in the future.

nicholasdaijh commented on src/main/java/woof/ui/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     * @param task        The task that was deleted.
+     * @param totalTasks  The total number of tasks after deleting the task.
+     */
+    public void displayTaskDeleted(String task, int totalTasks) {
```
should use public void displayTasksDeleted instead of displayTaskDeleted to follow convention, just like how you use displayTasksCleared instead of displayTaskCleared

nicholasdaijh commented on src/main/java/woof/ui/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * @param task The task that was unmarked.
+     */
+    public void displayTaskUnmarked(String task) {
```
should use public void displayTasksUnmarked instead of displayTaskUnmarked to follow convention, just like how you use displayTasksCleared instead of displayTaskCleared

nicholasdaijh commented on src/main/java/woof/ui/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * @param task The task that was marked as done.
+     */
+    public void displayTaskMarked(String task) {
```
should use public void displayTasksMarked instead of displayTaskMarked to follow convention, just like how you use displayTasksCleared instead of displayTaskCleared

nicholasdaijh commented on src/main/java/woof/ui/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     * @param task        The task that was added.
+     * @param totalTasks  The total number of tasks after adding the new task.
+     */
+    public void displayTaskAdded(String task, int totalTasks) {
```
should use public void displayTasksAdded instead of displayTaskAdded to follow convention, just like how you use displayTasksCleared instead of displayTaskCleared

nicholasdaijh reviewed (commented):
Overall well done!!

Commit:
Add Javadoc comments to methods

Commit:
Adhere to coding standards

Commit:
Adhere to coding standards for static

Commit:
Add find method

Commit:
Add add description method to task

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Resolve merge conflicts

andong0909 commented:
> Overall well done!!

Thanks Nicholas!

Commit:
Add checkstyle

hengjeekuan commented on src/main/java/woof/parser/Parser.java:
```diff
@@ -0,0 +1,170 @@
+import woof.command.*;
+import woof.task.TaskList;
+
+public class Parser {
```
LGTM

hengjeekuan commented on src/main/java/woof/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
```
LGTM

hengjeekuan commented on src/main/java/woof/task/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+        try {
+            this.deadline = LocalDate.parse(deadline);
+        } catch (DateTimeParseException e) {
+            throw new IllegalDateException("");
```
Is the message here supposed to be empty?

hengjeekuan commented on src/main/java/woof/task/Event.java:
```diff
@@ -0,0 +1,53 @@
+     *
+     * @return The description of the task.
+     */
+    public String toString() {
+        return "[E]" + super.toString() + " (from: "
+                + from.format(DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm")) + " to: "
+                + to.format(DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm")) + ")";
+    }
```
I think you're missing an @Override here

hengjeekuan commented on src/main/java/woof/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     *
+     * @return The description of the task.
+     */
+    public String print() {
```
I think you're missing an @Override here.

hengjeekuan commented on src/main/java/woof/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     * @param task Description of the task.
+     */
+    public Task(String task) {
+        this.task = task;
```
I think the naming of task here can be more intuitive, perhaps 'description'?

Commit:
Add gui using javafx

Commit:
Add assertion checks

Assertion checks are added to key lines to document
important assumptions, such as null checks, and array lengths.

This helps to ensure that the program is functioning as expected,
and that the code is robust against unexpected inputs.

Commit:
Adhere to coding standard

Codes are refactored to adhere to the given coding standard.

Commit:
Adhere to coding standard

Codes are refactored to adhere to the given coding standard.

andong0909 committed:
Merge pull request #1 from andong0909/branch-A-Assertions

Add assertion checks

Commit:
Merge branch 'master' into branch-A-CodeQuality

andong0909 committed:
Merge pull request #2 from andong0909/branch-A-CodeQuality

Branch a code quality

Commit:
Add continuous integration

Commit:
Add help feature

Commit:
Refine user guide

Commit:
Refine user guide

Commit:
Declare ai usage

Commit:
Adhere to coding standard

Commit:
Debug delete command

Commit:
Debug exit command

Commit:
Debug welcome message

Commit:
Adhere to coding standard

Commit:
Update user guide
